### ISSUE TITLE
Is it still not out of the gradient chain when calculating c?

### ISSUE THREAD

**Issue by coderhss** [IBM/online-alt-min#1]

**Comment by irinarish** [comment]
It is still out of chain;   the gradient of c is computed, but its value is not used directly to compute another gradient, since the optimization of c is a layer-local problem. Once c is updated locally, and fixed, the next layer c will be now optimized. Again, locally,  computing its gradient. But not involving the gradient of previous c. Or gradients of the weigjts.

**Comment by coderhss** [comment]
`loss = (1/mu)*loss_fn(output) + F.mse_loss(codes, codes_targets)`
the second term of the loss function always equal to zero?

**Comment by matrig** [comment]
No. `loss` is a function of `codes` and it is being optimized by finding a `codes` variable that minimizes it, i.e. minimizes the sum of the two terms. Please, see the paper for further clarifications.
